libraries/datadogstatsd.php

```diff
@@ -103,97 +108,107 @@ public static function decrement($stats, $sampleRate = 1, array $tags = null) {
-        	
             $data[$stat] = "$delta|c";
-			
         }
-
         static::send($data, $sampleRate, $tags);
-		
     }
 
     /**
      * Squirt the metrics over UDP
      * @param array $data Incoming Data
      * @param float|1 $sampleRate the rate (0-1) for sampling.
-	 * @param array|string $tags Key Value array of Tag => Value, or single tag as string
-	 * 
+     * @param array|string $tags Key Value array of Tag => Value, or single tag as string
+     *
      * @return null
      **/
     public static function send($data, $sampleRate = 1, array $tags = null) {
-
         // sampling
         $sampledData = array();
-
-		if ($sampleRate < 1) {
-        	
+        if ($sampleRate < 1) {
             foreach ($data as $stat => $value) {
-            	
                 if ((mt_rand() / mt_getrandmax()) <= $sampleRate) {
-                	
                     $sampledData[$stat] = "$value|@$sampleRate";
-					
-				}
-				
-			}
-			
-		} else {
-        	
+                }
+            }
+        } else {
             $sampledData = $data;
-			
-		}
+        }
 
         if (empty($sampledData)) { return; }
-	
-        foreach ($sampledData as $stat => $value) {
-
-			if ($tags !== NULL && is_array($tags) && count($tags) > 0) {
-				
-				$value .= '|';
-				
-				foreach ($tags as $tag_key => $tag_val) {
-					
-					$value .= '#' . $tag_key . ':' . $tag_val . ',';
-					
-				}
-				
-				$value = substr($value, 0, -1);
-				
-			} elseif (isset($tags) && !empty($tags)) {
-				
-				$value .= '|#' . $tags;
-				
-			}
 
+        foreach ($sampledData as $stat => $value) {
+            if ($tags !== NULL && is_array($tags) && count($tags) > 0) {
+                $value .= '|';
+                foreach ($tags as $tag_key => $tag_val) {
+                    $value .= '#' . $tag_key . ':' . $tag_val . ',';
+                }
+                $value = substr($value, 0, -1);
+            } elseif (isset($tags) && !empty($tags)) {
+                $value .= '|#' . $tags;
+            }
             static::report_metric("$stat:$value");
+        }
+    }
+    /**
+     * Send a custom service check status over UDP
+     * @param string $name service check name
+     * @param int $status service check status code (see static::OK, static::WARNING,...)
+     * @param array|string $tags Key Value array of Tag => Value, or single tag as string
+     * @param int $timestamp timestamp for the service check status (defaults to now)
+     * @param string $hostname hostname to associate with this service check status
+     * @param string $message message to associate with this service check status
+     *
+     * @return null
+     **/
+    public static function service_check($name, $status, array $tags = null,
+                                        $hostname = null, $message = null, $timestamp = null) {
+        $msg = "_sc|$name|$status";
 
-		}
+        if ($timestamp !== null) {
+            $msg .= sprintf("|t:%s", $timestamp);
```

Comment:
Looks like the correct prefix to modify the timestamp is `d:`
see https://github.com/DataDog/dd-agent/blob/master/aggregator.py#L564-L565